### PR TITLE
Updating the way we display rules on the ingress list and detail pages

### PR DESCRIPTION
**Pull request by codyrancher**
-Accounts for undefined hosts and paths
 -Accounts for wildcards in hosts
 -Accounts for protocol+host+paths not being a url
 -Removes links from paths on the detail page

 rancher/dashboard#762

### CODE REVIEW

File: components/formatter/IngressTarget.vue [hunk 1]
```diff
@@ -63,7 +63,7 @@ export default {
         const paths = rule?.http?.paths || [];
 
         return paths.map(path => ({
-          url:             `${ rule.host }${ path.path }`,
+          url:             `${ rule.host }${ path.path || '' }`,
```
Comment (mantis-toboggan-md) [inline components/formatter/IngressTarget.vue:66]: It looks like the same thing will happen with `rule.host` if it's undefined here

**Comment by vincent99** [comment]
Yes the host doesn't have to be set.

There's related problems here, not everything this is spitting out should be turned into a clickable `<a>`.  The host can be missing or contain wildcards.  The path can be a regex or contain other funky chars if `pathType`=`ImplementationSpecific` (which is the default, and not presented as an option at all today).  None of those are going to result in a useful clickable link so they should be just text.  And if there's `tls` hosts configured the link should go to http instead of https.

**Comment by codyrancher** [comment]
I've update it to account for most of what you've listed by making it more consistent with rancher. I do account for the wildcard in the host on the list page even though rancher does not right now. I've also filed a separate bug for the detail top as it currently doesn't have any content rancher/dashboard#805

Screen Shot 2020-07-02 at 4 03 57 PM
Screen Shot 2020-07-02 at 4 03 37 PM

**Comment by codyrancher** [comment]
@vincent99 would you mind taking a look at this one again?

File: edit/networking.k8s.io.ingress/Rules.vue [hunk 2]
```diff
@@ -76,55 +78,26 @@ export default {
...
-      });
-
-      return rows;
-    },
   }
 };
 </script>
 
 <template>
   <div v-if="isView">
```
[the remark this answers is not in the document]
Comment (codyrancher) [reply edit/networking.k8s.io.ingress/Rules.vue:86]: Done.

File: components/formatter/IngressTarget.vue [hunk 3]
```diff
@@ -19,90 +19,25 @@ export default {
...
-      };
-    }
-
-    return {
-      name, params, targetsWorkload
-    };
-  },
   computed: {
     workloads() {
       return Object.values(WORKLOAD_TYPES).flatMap(type => this.$store.getters['cluster/all'](type));
```
[the remark this answers is not in the document]
Comment (codyrancher) [reply components/formatter/IngressTarget.vue:24]: I'm not sure what you're getting at here. Would you mind elaborating?

[the remark this answers is not in the document]
Comment (vincent99) [reply components/formatter/IngressTarget.vue:24]: GitHub seems to have lost the line pointer.. but for the computed `workloads`, if a new workload is added from a change event, does it recompute here and get added to the list?

**Pull request merged**